saltproc/depcode.py

```diff
@@ -159,20 +166,346 @@ def write_mat_file(self, dep_dict, dep_end_time):
+        # Need to figure out how to adapt this to openmc
+        try:
+            subprocess.check_output(
+                args,
+                cwd=os.path.split(self.template_inputfile_path)[0],
+                stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as error:
+            print(error.output.decode("utf-8"))
+            raise RuntimeError('\n %s RUN FAILED\n see error message above'
+                               % (self.codename))
+        print('Finished OpenMC Run')
+
+    def switch_to_next_geometry(self):
+        """Switches the geometry file for the OpenMC depletion simulation to
+        the next geometry file in `geo_files`.
+        """
+        mats = openmc.Materials.from_xml(self.iter_matfile)
+        next_geometry = openmc.Geometry.from_xml(
+            path=self.geo_files.pop(0),
+            materials=mats)
+        next_geometry.export_to_xml(path=self.iter_inputfile['geometry'])
+        del mats, next_geometry
+
+    def write_depcode_input(self, reactor, dep_step, restart):
+        """ Writes prepared data into OpenMC input file(s).
+
+        Parameters
+        ----------
+        reactor : Reactor
+            Contains information about power load curve and cumulative
+            depletion time for the integration test.
+        dep_step : int
+            Current depletion time step.
+        restart : bool
+            Is the current simulation restarted?
+        """
+
+        if dep_step == 0 and not restart:
+            materials = openmc.Materials.from_xml(
+                self.template_inputfiles_path['materials'])
+            geometry = openmc.Geometry.from_xml(
+                self.template_inputfiles_path['geometry'], materials=materials)
+            settings = openmc.Settings.from_xml(
+                self.template_inputfiles_path['settings'])
+            settings.particles = self.npop
+            # settings.generations_per_batch = ??
+            settings.inactive = self.inactive_cycles
+            settings.batches = self.active_cycles + self.inactive_cycles
+        else:
+            materials = openmc.Materials.from_xml(self.iter_matfile)
+            geometry = openmc.Geometry.from_xml(
+                self.iter_inputfile['geometry'], materials=materials)
+            settings = openmc.Settings.from_xml(
+                self.iter_inputfile['settings'])
+
+        materials.export_to_xml(self.iter_matfile)
+        geometry.export_to_xml(self.iter_inputfile['geometry'])
+        settings.export_to_xml(self.iter_inputfile['settings'])
+        self.write_depletion_settings(reactor, dep_step)
+        self.write_saltproc_openmc_tallies(materials, geometry)
+        del materials, geometry, settings
+
+    def write_depletion_settings(self, reactor, current_depstep_idx):
+        """Write the depeletion settings for the ``openmc.depelete``
+        module.
+
+        Parameters
+        ----------
+        reactor : Reactor
```

Comment:
Perhaps Reactor object

Reply:
I think just `Reactor` here is appropriate, since that is the type of the object `reactor`

Reply:
actually I changed my mind and now agree with you